app/main/posts/modify/post-data-editor.directive.js

```diff
@@ -415,19 +429,24 @@ function PostDataEditorController(
             }
             request.$promise.then(function (response) {
                 var success_message = (response.status && response.status === 'published') ? 'notify.post.save_success' : 'notify.post.save_success_review';
+
                 // Save the updated post back to outside context
-                $scope.postContainer.post = response;
+                $scope.postContainer.post = $scope.post;
+
+                // DEVNOTE: Not sure how this would ever happen in the case of data view
```

Comment:
This handling exists because anonymous post submissions receive a 204 with no content, so no post id. Might be worth noting that in a comment

Reply:
This is not relevant to the data-editor though since it isn't used to create.